components/Button/Button.jsx

```diff
@@ -1,91 +1,329 @@
+    };
+    return `font-size: ${sizes[size] || sizes.medium};`;
+  }}
+
+  ${({
+    size,
+    theme: {
+      spacing: { large, xlarge, xxlarge, xxxlarge },
+    },
+  }) => {
+    const heights = {
+      xsmall: `${large}px`,
+      small: `${xlarge}px`,
+      medium: `${xxlarge}px`,
+      large: `${xxxlarge}px`,
+    };
+
+    return `height: ${heights[size] || heights.medium};`;
+  }}
+
+  ${({
+    size,
+    theme: {
+      spacing: { small, medium },
+    },
+  }) => {
+    const paddings = {
+      xsmall: `0 ${small}px`,
+      small: `0 ${small}px`,
+      medium: `0 ${medium}px`,
+      large: `0 ${medium}px`,
+      xlarge: `0 ${medium}px`,
+    };
+
+    return `padding: ${paddings[size] || paddings.medium};`;
+  }}
+
+  ${props =>
+    props.center &&
+    `
+    margin-left: auto;
+    margin-right: auto;
+  `}
 
   ${ButtonIcon} {
-    ${iconSize}
+    margin-right: 5px;
   }
+
+  transition: all 0.2s ease-in-out;
+
+  ${props => props.full && `width: 100%;`}
+
+  ${({ skin, theme, disabled, stroked }) => {
+    const {
+      components: {
+        button: {
+          skins: {
+            [skin]: {
+              mainColor: {
+                100: mainColor100,
+                500: mainColor500,
+                700: mainColor700,
+              },
+              text: { 100: text100 },
+            },
+          },
+        },
+      },
+      colors: {
+        neutral: { 100: neutral100, 500: neutral500 },
+      },
+    } = theme;
+
+    let bgColor;
+    let textColor;
+
+    if (disabled && stroked) {
+      bgColor = neutral100;
+      textColor = neutral500;
+    } else if (stroked) {
+      bgColor = neutral100;
+      textColor = mainColor500;
+    } else if (disabled) {
+      bgColor = neutral500;
+      textColor = neutral100;
+    } else {
+      bgColor = mainColor500;
+      textColor = text100;
+    }
+
+    return `
+      background-color: ${bgColor};
+      color: ${textColor};
+
+      border: 1.5px solid ${disabled ? neutral500 : mainColor500};
+
+      ${shadow(2, neutral500)({ theme })}
+
+      :hover {
+        ${
+          !disabled
+            ? `
+              ${shadow(4, mainColor700)({ theme })}
+              background-color: ${stroked ? mainColor100 : mainColor700};
+              border-color: ${mainColor700};
+            `
+            : ''
+        }
+      }
+
+      :focus,
+      :focus-within {
+        ${!disabled ? shadow(4, mainColor500)({ theme }) : ''}
+      }
+
+      :active {
+        ${
+          !disabled
+            ? `
+              ${shadow(8, mainColor700)({ theme })}
+              background-color: ${stroked ? mainColor100 : mainColor700};
+            `
+            : ''
+        }
+      }
+
+    ${ButtonIcon} {
+      ${({
+        size,
+        theme: {
+          spacing: { medium, large },
+        },
+      }) => {
+        const sizes = {
+          xsmall: `${medium}px`,
+          small: `${medium}px`,
+          normal: `${large}px`,
```

Comment:
I guess we can update this object and rename `normal` to `medium` to keep with our nomenclature convention

Reply:
:+1: